doc/DataPlaneCodeGeneration/Handle_Ambiguity.md

```diff
@@ -0,0 +1,228 @@
+# Handle Ambiguity Between Protocol Method and Convenience Method
+
+## Background
+
+In a DPG library, protocol methods will be generated for a swagger input, and both of protocol methods and convenience methods will be generated for a typespec input. We expect to make convenience method an overload of protocol method, so that users could be very clear they are representing the same operation. For example,
+
+```C#
+public virtual async Task<Response<Model>> OperationAsync(string requiredId, string optionalId = null, CancellationToken cancellationToken = default) // Convenience method
+{
+    ...
+}
+
+public virtual async Task<Response> OperationAsync(string requiredId, string optionalId = null, RequestContext context = null) // Protocol method
+{
+    ...
+}
+```
+However, this leads to compilation error
+```
+Error	CS0121	The call is ambiguous between the following methods or properties: 'SomeClass.OperationAsync(string, CancellationToken)' and 'SomeClass.OperationAsync(string, RequestContext)'
+```
+when calling `OperationAsync("<some string>")`.
+
+## Solution
+
+There are two ways to handle this error:
+
+1. Make all the parameters in **protocol methods** to required.
+```C#
+public virtual async Task<Response<Model>> OperationAsync(string requiredId, string optionalId = null, CancellationToken cancellationToken = default) // Convenience method
+{
+    ...
+}
+
+public virtual async Task<Response> OperationAsync(string requiredId, string optionalId, RequestContext context) // Protocol method
+{
+    ...
+}
+```
+2. Append `Value` to name of **convenience methods**.
+```C#
+public virtual async Task<Response<Model>> OperationValueAsync(string requiredId, string optionalId = null, CancellationToken cancellationToken = default) // Convenience method
+{
+    ...
+}
+
+public virtual async Task<Response> OperationAsync(string requiredId, string optionalId = null, RequestContext context = null) // Protocol method
+{
+    ...
+}
+```
+
+The first solution is our preference, so when codegen detects the existing of ambiguity, it chooses the first strategy to resolve the problem.
+
+Only when protocol method already GAed so that we cannot change it, the codegen will choose the second strategy.
+
+TO-DO: I actually don't know the difference between these two flags.
```

Comment:
Can we resolve this before you merge?

Reply:
I removed and created an issue for this https://github.com/Azure/azure-sdk-for-net/issues/38405